numpy/ma/core.py

```diff
@@ -901,22 +901,32 @@ class _MaskedBinaryOperation:
         Filling value for the first argument, default is 0.
     filly : scalar, optional
         Filling value for the second argument, default is 0.
-
+    op    : bool, optional
+        Set to true when used in (non-reversed) binary operators.
     """
-    def __init__ (self, mbfunc, fillx=0, filly=0):
+    def __init__ (self, mbfunc, fillx=0, filly=0, op=False):
         """abfunc(fillx, filly) must be defined.
            abfunc(x, filly) = x for all x to enable reduce.
         """
         self.f = mbfunc
         self.fillx = fillx
         self.filly = filly
+        self.op = op
         self.__doc__ = getattr(mbfunc, "__doc__", str(mbfunc))
         self.__name__ = getattr(mbfunc, "__name__", str(mbfunc))
         ufunc_domain[mbfunc] = None
         ufunc_fills[mbfunc] = (fillx, filly)
 
+    def __get__(self, obj, type=None):
+        return partial(self, obj)
+
     def __call__ (self, a, b, *args, **kwargs):
         "Execute the call behavior."
+        # Check array priority
+        if self.op:
```

Comment:
In `ndarray` C code, `NotImplemented` is only returned if the array priority is higher *and* the other class defines the reverse operation. I think one could use a similar structure as you have here, except use `rop` to pass on the reverse operator and here do:
```
if self.rop and getattr(b, self.rop, None) and getattr(b, '__array_priority__', -1) > a.__array_priority__:
    return NotImplemented
```